lib/phoenix_live_view/router.ex

```diff
@@ -365,6 +366,69 @@ defmodule Phoenix.LiveView.Router do
     end
   end
 
+  @doc """
+  Applies Liveview session data.
+
+  This plug must be placed **after** the `:protect_from_forgery` plug.
+  """
+  def fetch_liveview_session(conn, _opts \\ []) do
+    {conn, data} = get_put_session_cookie(conn)
+
+    conn =
+      case data do
+        %{csrf_token: csrf_token, ops: ops} ->
+          conn
+          |> verify_csrf!(csrf_token)
+          |> apply_session_ops(ops)
+
+        _ ->
+          conn
+      end
+
+    conn = Plug.Conn.put_private(conn, :lv_put_session_ref, make_ref())
+
+    # This is needed for dead live_render in controllers,
+    # because those only return the rendered content and not the conn.
+    #
+    # We also use this for regular dead mounts to simplify the code.
+    Plug.Conn.register_before_send(conn, fn conn ->
```

Comment:
I don't understand why we have to postpone the changing of the session, let's sync later!